server/controllers/recommendation/editDateOfBirthController.test.ts

```diff
@@ -16,8 +16,11 @@ describe('get', () => {
     const res = mockRes({
       locals: {
         recommendation: {
-          bookRecallToPpud: { dateOfBirth: '2003-01-01' },
+          prisonOffender: { dateOfBirth: '2003-01-01' },
+          ppudOffender: { dateOfBirth: '1984-09-08' },
         },
+        dateOfBirthNomis: '2003-01-01',
+        dateOfBirthPpud: '1984-09-08',
```

Comment:
Are you sure you want to do this?  You set the value that you assert on later.  The controller could do nothing with `res` and the test would pass, I think.

Reply:
The recommendation is of course always present on `res.locals` in the recommendation route.
`res.locals.recommendation`.
>The controller could do nothing with res and the test would pass, I think.

I used the controller to grab a couple of fields from the recommendation and add two new `res.locals` fields
`dateOfBirthNomis` and `dateOfBirthPpud`. 
The test asserts that those two fields are present an has the correct value. If I removed `dateOfBirthNomis` and `dateOfBirthPpud` from `res.locals` in the controller the test will fail.

Reply:
OK. So don't set the values in the test then.

Reply:
The real `res.locals` receives the recommendation from the previous page in the rec flow. The `res.locals` in the test is a mock that needs to have it's values set.

Reply:
Done ... or rather undone! With a bit of a refactor! 👍